src/components/button/button-theme.scss

```diff
@@ -5,7 +5,7 @@ a.md-button.md-THEME_NAME-theme,
     &:hover {
       background-color: '{{background-500-0.2}}';
     }
-    &.md-focused {
+    &.md-focused:not(.md-no-focus) {
```

Comment:
Seems to be a better a idea, I would prefer a attribute called `md-no-focus-style`. See my newest changes, I updated my PR, you should decide about `className` or `attribute`